java-checks/src/main/java/org/sonar/java/symexecengine/State.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * dev@sonar.codehaus.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.symexecengine;
+
+import com.google.common.collect.Lists;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import java.util.List;
+
+public abstract class State {
+  private final List<Tree> changingStateTrees;
+
+  public static State UNSET = new State() {
+    @Override
+    public State merge(State s) {
+      return s;
+    }
+  };
+
+  private State() {
+    changingStateTrees = Lists.newArrayList();
+  }
+
+  public State(Tree tree) {
+    changingStateTrees = Lists.newArrayList();
+    changingStateTrees.add(tree);
+  }
+
+  public State(List<Tree> trees) {
+    changingStateTrees = trees;
+  }
+
+  public abstract State merge(State s);
+
+  public boolean shouldRaiseIssue() {
+    return false;
+  }
+
+  public List<Tree> reportingTrees() {
+      return changingStateTrees;
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Make this line start at column 5.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)